accounts-db/src/accounts_db.rs

```diff
@@ -3994,7 +3988,7 @@ impl AccountsDb {
             let (shrink_in_progress, time_us) = measure_us!(
                 self.get_store_for_shrink(slot, shrink_collect.alive_total_bytes as u64)
             );
-            stats_sub.create_and_insert_store_elapsed_us = time_us;
+            stats_sub.create_and_insert_store_elapsed_us += time_us;
```

Comment:
Good call out - I probably should have included a note about this. This was actually something I learned as well; note the [following](https://doc.rust-lang.org/std/ops/):

> the assignment operator (=) has no backing trait, there is no way of overloading its semantics

The `lhs` is a `std::num::Saturating<usize>`, `rhs` is a `usize` so we can't use `=` here. Given that we just created `stats_sub` above with `ShrinkStatsSub::default()`, switching to use `+=` seemed fine from a functionality point of view. The alternative would be:
```rust
stats_sub.create_and_insert_store_elapsed_us = Saturating(time_us);
```

That being said, given that `std::num::Saturating` is `repr(transparent)`, doing the assignment is probably cheaper and godbolt agrees with me so I'll update all these:
https://godbolt.org/z/vfhPr9daT

Reply:
They should be the same.
If you turn on the optimizations, both are indeed compiled to
```asm
        mov     eax, edi
        ret
```
https://godbolt.org/z/EoToT33j1

Personally, I do not have a strong preference between `= Saturating(v)` vs `+= v`.
In this particular case, the initial value is holding all zeroes, so `+=` seems fine.

I am again surprised by the lack of ergonomics for `Saturating` :(

Reply:
Changed in https://github.com/anza-xyz/agave/pull/523/commits/a1e70527a0c1b04ec09a0dae7b5684b596dc6848

Reply:
Ahh yes, good call on flipping on optimizations

> Personally, I do not have a strong preference between = Saturating(v) vs += v.

Me neither, `= Saturating(v)` more closely matches original behavior so that seems like a reasonable tiebreaker

> In this particular case, the initial value is holding all zeroes, so += seems fine.

Exactly, this was my original justification for switch to `+=` to be fine.

> I am again surprised by the lack of ergonomics for Saturating

Ya, I plan on making a PR to Rust in the next week or two when I get a chance; several lines of (simple) could seemingly improve ergonomics:
- Eq / PartialEq / PartialOrd between `T` and `Saturating<T>`
    - This will avoid having to do `.0` on the `Saturating` operand
- From / Into also seem like good candidates